ci/src/main.rs

```diff
@@ -0,0 +1,213 @@
+
+            debug("Build and test individual features along with \"std\":");
+            for feature in FEATURES.split(" ") {
+                let features = format!("std {}", feature);
+                if args.debug {
+                    eprintln!("\t feature \"{}\"", feature);
+                }
+                run_cmd!(cargo test --no-default-features --features="$features")?;
+            }
+
+            debug("Build std + no_std, to make sure they are not incompatible ...");
+            run_cmd!(cargo build --no-default-features --features="std no-std")?;
+        }
+
+        debug("Run the examples with the \"std\" feature enabled ...\n");
+        run_cmd!(cargo run --example bip32 7934c09359b234e076b9fa5a1abfd38e3dc2a9939745b7cc3c22a48d831d14bd)?;
+
+    } else {
+        debug("Build no_std, to make sure that cfg(test) doesn't hide any issues ...");
+        run_cmd!(cargo build --no-default-features --features="no-std")?;
+
+        debug("Test with no features other than \"no-std\" ...");
+        run_cmd!(cargo test --no-default-features --features="no-std")?;
+
+        debug("Build all the features excluding \"std\" ...");
+        let features = format!("no-std {}", FEATURES);
+        run_cmd!(cargo test --no-default-features --features="$features")?;
+
+        if args.feature_matrix {
+            debug("Build and test individual features along with \"no-std\" ...");
+            for feature in FEATURES.split(" ") {
+                let features = format!("no-std {}", feature);
+                run_cmd!(cargo test --no-default-features --features="$features")?;
+            }
+        }
+
+        debug("Run the examples without \"std\" and with \"no-std\"  ...");
+        run_cmd!(cargo run --no-default-features --features no-std --example bip32 7934c09359b234e076b9fa5a1abfd38e3dc2a9939745b7cc3c22a48d831d14bd)?;
+    }
+
+    if args.docs {
+        if is_stable_toolchain {
+            eprintln!("docs build cannot be doe with a stable toolchain");
+            process::exit(1);
+        }
+        debug("Building the docs ...");
+        run_cmd!(RUSTDOCFLAGS="--cfg docsrs" cargo doc --features="$FEATURES")?;
+    }
+
+    if args.fuzz {
+        debug("Running fuzz tests ...");
+        run_cmd!(cd fuzz; cargo test; ./travis-fuzz.sh)?;
+    }
+
+    if args.bench {
+        if is_stable_toolchain {
+            eprintln!("bench marks cannot be run with a stable toolchain");
+            process::exit(1);
+        }
+        debug("Running bench marks ...");
+        std::env::set_var("RUSTFLAGS", "--cfg=bench");
+        run_cmd!(cargo bench)?;
+    }
+
+    if args.as_dependency {
+        debug("\nTesting bitcoin can be used as a dependency ...");
+        let dep = "bitcoin = { path = \"..\", features = [\"serde\"] }";
+
+        run_cmd!(cargo new dep_test; echo "$dep" >> dep_test/Cargo.toml; cargo test --manifest-path dep_test/Cargo.toml)?;
```

Comment:
If the first command fails, the rest will continue doing funny things. `&&` is fine for quick fix but more nicely append the text using Rust.

Reply:
Good point, will do. Thanks